### PR TITLE
Update raja - chai - vtk - fmt, cpp17 standard and modify images to use cuda11 and cmake 3.23

### PR DESCRIPTION
**Pull request by CusiniM**
GEOS PR: https://github.com/GEOS-DEV/GEOS/pull/2496

### CODE REVIEW

File: docker/TotalEnergies/pecan-GPU.cmake [hunk 1]
```diff
@@ -3,13 +3,11 @@ include(${CMAKE_CURRENT_LIST_DIR}/pecan-CPU.cmake)
...
 set(ENABLE_CUDA ON CACHE PATH "" FORCE)
-set(CUDA_TOOLKIT_ROOT_DIR /hrtc/apps/cuda/10.2.89/x86_64 CACHE PATH "")
+set(CUDA_TOOLKIT_ROOT_DIR /hrtc/apps/cuda/11.2.0/x86_64 CACHE PATH "")
 set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER} CACHE STRING "")
 set(CMAKE_CUDA_COMPILER ${CUDA_TOOLKIT_ROOT_DIR}/bin/nvcc CACHE STRING "")
 
 set(CUDA_ARCH sm_75 CACHE STRING "")
-set(CMAKE_CUDA_STANDARD 14 CACHE STRING "")
-### The inclusion of -std=c++14 is a workaround for a cuda10/gcc8 bug ###
 set(CMAKE_CUDA_FLAGS "-restrict -arch ${CUDA_ARCH} --expt-relaxed-constexpr --expt-extended-lambda -Werror cross-execution-space-call,reorder,deprecated-declarations -Xcompiler -std=c++14" CACHE STRING "")
```
Comment (CusiniM) [inline docker/TotalEnergies/pecan-GPU.cmake:11]:
```suggestion
set(CMAKE_CUDA_FLAGS "-restrict -arch ${CUDA_ARCH} --expt-relaxed-constexpr --expt-extended-lambda -Werror cross-execution-space-call,reorder,deprecated-declarations -Xcompiler" CACHE STRING "")
```

**Review by rrsettgast — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -306,8 +311,9 @@ list(APPEND build_list silo )
 
 
 set(RAJA_DIR "${CMAKE_INSTALL_PREFIX}/raja")
-#set(RAJA_URL "${TPL_MIRROR_DIR}/RAJA-v2022.03.0.tar.gz")
-set(RAJA_URL "${TPL_MIRROR_DIR}/RAJA-feature-cmake-config-reorg.tar.gz")
+#set(RAJA_URL "${TPL_MIRROR_DIR}/RAJA-v2022.10.5.tar.gz")
+#set(RAJA_URL "${TPL_MIRROR_DIR}/raja-c1ff3a6e0e3599be4174a89283983297474f1ddc.tar.gz")
```
Comment (rrsettgast) [inline CMakeLists.txt:315]:
```suggestion
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -345,7 +351,8 @@ list(APPEND build_list raja )
 # CHAI
 ################################
 set(CHAI_DIR "${CMAKE_INSTALL_PREFIX}/chai")
-set(CHAI_URL "${TPL_MIRROR_DIR}/chai-2022.03.0.tar.gz")
+#set(CHAI_URL "${TPL_MIRROR_DIR}/chai-36297e526706a805957b3fb77425cec1de864227.tar.gz")
```
Comment (rrsettgast) [inline CMakeLists.txt:354]:
```suggestion
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -957,7 +968,8 @@ prefix    = ${SCOTCH_DIR}
     file( WRITE ${PROJECT_BINARY_DIR}/scotch_makefile.inc "${SCOTCH_CONFIG}")
     set( SCOTCH_BUILD_DIR "${PROJECT_BINARY_DIR}/scotch/src/scotch/src" )
     if ( NOT DEFINED SCOTCH_NUM_PROC )
-      set( SCOTCH_NUM_PROC ${NUM_PROC} )
+#      set( SCOTCH_NUM_PROC ${NUM_PROC} )
+      set( SCOTCH_NUM_PROC 2 )
```
Comment (rrsettgast) [inline CMakeLists.txt:972]:
```suggestion
      set( SCOTCH_NUM_PROC ${NUM_PROC} )
```

File: docker/TotalEnergies/pecan-GPU.cmake [hunk 5]
```diff
@@ -3,14 +3,13 @@ include(${CMAKE_CURRENT_LIST_DIR}/pecan-CPU.cmake)
...
+set(CUDA_TOOLKIT_ROOT_DIR /hrtc/apps/cuda/11.5.119/x86_64/centos7 CACHE PATH "")
 set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER} CACHE STRING "")
 set(CMAKE_CUDA_COMPILER ${CUDA_TOOLKIT_ROOT_DIR}/bin/nvcc CACHE STRING "")
 
 set(CUDA_ARCH sm_75 CACHE STRING "")
-set(CMAKE_CUDA_STANDARD 14 CACHE STRING "")
-### The inclusion of -std=c++14 is a workaround for a cuda10/gcc8 bug ###
-set(CMAKE_CUDA_FLAGS "-restrict -arch ${CUDA_ARCH} --expt-relaxed-constexpr --expt-extended-lambda -Werror cross-execution-space-call,reorder,deprecated-declarations -Xcompiler -std=c++14" CACHE STRING "")
+set(CMAKE_CUDA_ARCHITECTURES 75 CACHE STRING "")
+set(CMAKE_CUDA_FLAGS "-restrict -arch ${CUDA_ARCH} --expt-relaxed-constexpr --expt-extended-lambda -Werror cross-execution-space-call,reorder,deprecated-declarations" CACHE STRING "")
```
Comment (CusiniM) [inline docker/TotalEnergies/pecan-GPU.cmake:12]:
```suggestion
set(CMAKE_CUDA_FLAGS "-restrict -arch ${CUDA_ARCH} --expt-relaxed-constexpr --expt-extended-lambda -Werror cross-execution-space-call,reorder,deprecated-declarations -Xcompiler" CACHE STRING "")
```

File: docker/TotalEnergies/pecan-GPU.cmake [hunk 6]
```diff
@@ -3,14 +3,13 @@ include(${CMAKE_CURRENT_LIST_DIR}/pecan-CPU.cmake)
...
+set(CUDA_TOOLKIT_ROOT_DIR /hrtc/apps/cuda/11.5.119/x86_64/centos7 CACHE PATH "")
 set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER} CACHE STRING "")
 set(CMAKE_CUDA_COMPILER ${CUDA_TOOLKIT_ROOT_DIR}/bin/nvcc CACHE STRING "")
 
 set(CUDA_ARCH sm_75 CACHE STRING "")
-set(CMAKE_CUDA_STANDARD 14 CACHE STRING "")
-### The inclusion of -std=c++14 is a workaround for a cuda10/gcc8 bug ###
-set(CMAKE_CUDA_FLAGS "-restrict -arch ${CUDA_ARCH} --expt-relaxed-constexpr --expt-extended-lambda -Werror cross-execution-space-call,reorder,deprecated-declarations -Xcompiler -std=c++14" CACHE STRING "")
+set(CMAKE_CUDA_ARCHITECTURES 75 CACHE STRING "")
+set(CMAKE_CUDA_FLAGS "-restrict -arch ${CUDA_ARCH} --expt-relaxed-constexpr --expt-extended-lambda -Werror cross-execution-space-call,reorder,deprecated-declarations -Xcompiler" CACHE STRING "")
```
Comment (CusiniM) [inline docker/TotalEnergies/pecan-GPU.cmake:12]:
```suggestion
set(CMAKE_CUDA_FLAGS "-restrict -arch ${CUDA_ARCH} --expt-relaxed-constexpr --expt-extended-lambda -Werror cross-execution-space-call,reorder,deprecated-declarations" CACHE STRING "")
```

Comment (CusiniM) [reply docker/TotalEnergies/pecan-GPU.cmake:12]: @rrsettgast you can also commit that since you are modifying the docker files.

**Pull request merged**